### ISSUE TITLE
OASIS_infobyte_EMAIL SPAM DETECTION WITH MACHINE LEARNING

### ISSUE THREAD

**Issue by Shivharekanhaiya** [Shivharekanhaiya/kanhaiya-shivhare#7]
import tkinter as tk
from tkinter import messagebox
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.svm import SVC

# Load the email spam dataset
data = pd.read_csv('email_spam_dataset.csv')  # Replace with the actual path to your dataset file

# Preprocess the data
# ...

# Train the spam detection model
# ...

# Function to preprocess input email text and predict spam/ham
def predict_spam(email_text):
    # Preprocess the input email text (e.g., cleaning, transforming into numerical features)
    # ...
    
    # Perform the prediction using the trained model
    # ...
    
    # Return the predicted spam/ham label
    # ...

# Function to handle button click event
def detect_spam():
    # Get the input email text from the GUI widget
    # ...
    
    # Call the predict_spam() function to get the predicted label
    predicted_label = predict_spam(email_text)
    
    # Show a message box with the predicted label
    messagebox.showinfo("Spam Detection Result", f"Predicted Label: {predicted_label}")

# Create the main window
window = tk.Tk()
window.title("Email Spam Detection")

# Create input widgets for email text
# ...

# Create the detect button
detect_button = tk.Button(window, text="Detect Spam", command=detect_spam)
detect_button.pack(pady=10)

# Start the GUI event loop
window.mainloop()
